toplevel/opttoploop.ml

```diff
@@ -610,21 +610,25 @@ let loop ppf =
     | x -> Location.report_exception ppf x; Btype.backtrack snap
   done
 
-(* Execute a script.  If [name] is "", read the script from stdin. *)
-
-let override_sys_argv args =
-  let len = Array.length args in
-  if Array.length Sys.argv < len then invalid_arg "Toploop.override_sys_argv";
-  Array.blit args 0 Sys.argv 0 len;
-  Obj.truncate (Obj.repr Sys.argv) len;
+(* The script must see a different value for the "constant" Sys.argv.
+   So, rewrite the module to claim it was always that way *)
+module type SYS = module type of Sys
+
+[@@@ocaml.warning "-59"]
```

Comment:
Why are you disabling this warning here (59 Assignment to non-mutable value) and not in the toploop version?